rtos/source/TARGET_CORTEX/mbed_boot.c

```diff
@@ -64,6 +64,15 @@
 #include "mbed_error.h"
 #include "mbed_mpu_mgmt.h"
 
+#ifdef MBED_CONF_RTOS_PRESENT
+#include "cmsis_os2.h"
+#endif
+
+#ifdef MBED_CONF_RTOS_PRESENT
+extern osMutexId_t singleton_mutex_id;
```

Comment:
Wouldn't it be better to move the definitions (and initialisation of them) here? `mbed_rtos_rtx.c` is no longer using them, so odd to be defining in one place and using from another.

Or alternatively, less disruptively, have a `mbed_rtos_init_singleton_mutex()` function in mbed_rtos_rtx.c that does the init, declared in mbed_boot.h, you can call from here. Maybe that could be a more general name - something about "post-rtos-start" initialisation, but not sure what.

Another point while looking at this - the `singleton_mutex_attr` does not need to be in RAM, or even ROM. You could mark it `const` with an initialiser to get it out of RAM at least, or you could even just declare it locally in the function (on the stack) - it doesn't need to persist after the `osMutexNew`.  It's possible that having it locally may use more ROM in code than having it static and `const` uses in data.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/37d57137870fabf715fb6f294e2fb0b0703c9498..6d4a0cda9b19c833f1ca7cd8146a06119e459086).
The changes caused a decrease of 6 bytes in RAM and increase of 2 bytes in ROM.
Having `singleton_mutex_attr` locally does not use more ROM.